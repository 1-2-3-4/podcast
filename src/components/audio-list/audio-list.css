.track-item {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin: 10px;
  border-radius: 10px;
  background-color: #272727;
  padding: 20px;
  -webkit-box-shadow: inset -5px -5px 10px #202020, inset 5px 5px 10px #202020;
          box-shadow: inset -5px -5px 10px #202020, inset 5px 5px 10px #202020;
}

.track-item:hover {
  cursor: pointer;
  background-color: #202020;
}

.track-item-now-playing {
  background-color: #f2f2f2;
  -webkit-box-shadow: inset -5px -5px 10px #c2c2c2, inset 5px 5px 10px #c2c2c2;
          box-shadow: inset -5px -5px 10px #c2c2c2, inset 5px 5px 10px #c2c2c2;
  position: relative;
}

.track-item-now-playing:hover {
  background-color: #ffffff;
}

.track-text {
  margin-left: 15px;
  font-family: Helvetica, sans-serif;
}

.track-title {
  margin-bottom: 10px;
  font-weight: 700;
  font-size: 16px;
  color: #e2e2e2;
}

.track-title-now-playing {
  color: #565656;
}

.track-image {
  width: 50px;
}

.track-metadata {
  font-size: 12px;
  color: #9c9c9c;
}

.track-metadata-now-playing {
  color: #838383;
}