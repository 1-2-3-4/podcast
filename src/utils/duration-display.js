export const durationDisplay = (duration) => {
  const minutes = Math.floor((duration / (1000 * 60)) % 60)
  const seconds = Math.floor((duration / 1000) % 60)

  return `${minutes > 10 ? minutes : `0${minutes}`}:${seconds}`
}