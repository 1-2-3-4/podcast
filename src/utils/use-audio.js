import * as React from 'react'
import { AudioContext } from '../components/audio-provider/audio-context'

export const useAudio = () => React.useContext(AudioContext)