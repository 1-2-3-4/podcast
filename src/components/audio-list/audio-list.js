import * as React from 'react'
import cx from 'classnames'

import { durationDisplay } from '../../utils/duration-display'
import { useAudio } from '../../utils/use-audio'
import './audio-list.css'

export const AudioList = () => {
  const { tracks, nowPlayingIndex, setNowPlayingIndex } = useAudio()

  return (
    <>
      {tracks.map((track, i) => {
        const { duration, imageUrl, mediaUrl, title } = track
        const isNowPlaying = i === nowPlayingIndex

        return (
          <div
            className={cx(
              "track-item",
              {["track-item-now-playing"]: isNowPlaying}
            )}
            data-testid="track-item"
            key={mediaUrl}
            onClick={() => setNowPlayingIndex(i)}
          >
            <img className="track-image" src={imageUrl} alt={title} />
            <div className="track-text">
              <div
                className={cx(
                  "track-title",
                  {["track-title-now-playing"]: isNowPlaying}
                )}
              >
                {title}
              </div>
              <div
                className={cx(
                  "track-metadata",
                  {["track-title-now-playing"]: isNowPlaying}
                )}
              >
                {durationDisplay(duration)}
              </div>
            </div>
          </div>
        )
      })}
    </>
  )
}