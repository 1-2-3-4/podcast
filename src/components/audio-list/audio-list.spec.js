import * as React from 'react'
import { shallow } from 'enzyme'

import { AudioList } from './audio-list'
import * as useAudioUtil from '../../utils/use-audio'

describe('AudioList', () => {
  describe('when it renders', () => {
    let wrapper
    const mockTracks = [
      {
        title: 'This is a Fake Song',
        imageUrl: 'https://image.com/this-is-a-fake-song',
        mediaUrl: 'https://media.com/this-is-a-fake-song',
        duration: 10000
      },
      {
        title: 'Also a Mock Song',
        imageUrl: 'https://image.com/also-a-mock-song',
        mediaUrl: 'https://media.com/also-a-mock-song',
        duration: 10000
      },
      {
        title: 'Yup, Still More Fake Songs',
        imageUrl: 'https://image.com/yup-still-more-fake-songs',
        mediaUrl: 'https://media.com/yup-still-more-fake-songs',
        duration: 10000
      }
    ]
    const mockNowPlayingIndex = 2

    beforeAll(() => {
      jest.spyOn(useAudioUtil, 'useAudio').mockReturnValue({
        tracks: mockTracks,
        nowPlayingIndex: mockNowPlayingIndex,
      })

      wrapper = shallow(<AudioList/>)
    })

    afterAll(() => {
      jest.restoreAllMocks()
    })

    it('renders a track item for each track', () => {
      expect(wrapper.find({ 'data-testid': 'track-item' })).toHaveLength(mockTracks.length)
    })

    it('sets now-playing class on the currently playing track', () => {
      expect(wrapper.find({ 'data-testid': 'track-item' }).at(mockNowPlayingIndex).prop('className')).toContain('track-item-now-playing')
    })

    it('matches snapshot', () => {
      expect(wrapper.getElement()).toMatchSnapshot()
    })
  })
})
