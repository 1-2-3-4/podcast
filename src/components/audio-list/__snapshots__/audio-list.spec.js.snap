// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioList when it renders matches snapshot 1`] = `
<React.Fragment>
  <div
    className="track-item"
    data-testid="track-item"
    onClick={[Function]}
  >
    <img
      alt="This is a Fake Song"
      className="track-image"
      src="https://image.com/this-is-a-fake-song"
    />
    <div
      className="track-text"
    >
      <div
        className="track-title"
      >
        This is a Fake Song
      </div>
      <div
        className="track-metadata"
      >
        00:10
      </div>
    </div>
  </div>
  <div
    className="track-item"
    data-testid="track-item"
    onClick={[Function]}
  >
    <img
      alt="Also a Mock Song"
      className="track-image"
      src="https://image.com/also-a-mock-song"
    />
    <div
      className="track-text"
    >
      <div
        className="track-title"
      >
        Also a Mock Song
      </div>
      <div
        className="track-metadata"
      >
        00:10
      </div>
    </div>
  </div>
  <div
    className="track-item track-item-now-playing"
    data-testid="track-item"
    onClick={[Function]}
  >
    <img
      alt="Yup, Still More Fake Songs"
      className="track-image"
      src="https://image.com/yup-still-more-fake-songs"
    />
    <div
      className="track-text"
    >
      <div
        className="track-title track-title-now-playing"
      >
        Yup, Still More Fake Songs
      </div>
      <div
        className="track-metadata track-title-now-playing"
      >
        00:10
      </div>
    </div>
  </div>
</React.Fragment>
`;
